tests/memories_test.py

```diff
@@ -1305,13 +1305,10 @@ def f(x):
       self.assertRegex(compiled_text, r"copy-done.*S\(5\)")
 
     compiled_stats = compiled_f.memory_analysis()
-    if compiled_stats is not None and jtu.test_device_matches(["tpu"]):
```

Comment:
Can you please keep the `compiled_stats is not None` condition? This is needed for other clients that don't support compiled stats.

Reply:
Ouch, I am sorry. Done.